extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugInterface.java

```diff
@@ -70,22 +88,88 @@
             .build();
     this.accessToken = new Token(authData.getToken(), authData.getSecret());
     this.mapper = mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    this.user = getUserInformation().getUser();
   }
 
-  SmugMugResponse<SmugMugAlbumInfoResponse> makeAlbumInfoRequest(String url) throws IOException {
-    return makeRequest(url, new TypeReference<SmugMugResponse<SmugMugAlbumInfoResponse>>() {});
+  /* Returns the album information corresponding to the album URI provided. */
+  SmugMugAlbumInfoResponse getAlbumInfo(String url) throws IOException {
+    Preconditions.checkArgument(
+        !Strings.isNullOrEmpty(url), "Album URI is required to retrieve album information");
+    return makeRequest(url, new TypeReference<SmugMugResponse<SmugMugAlbumInfoResponse>>() {})
+        .getResponse();
   }
 
-  SmugMugResponse<SmugMugAlbumsResponse> makeAlbumRequest(String url) throws IOException {
-    return makeRequest(url, new TypeReference<SmugMugResponse<SmugMugAlbumsResponse>>() {});
+  /* Returns the album corresponding to the url provided. If the url is null or empty, this
+   * returns the top level user albums. */
+  SmugMugAlbumsResponse getAlbums(String url) throws IOException {
+    if (Strings.isNullOrEmpty(url)) {
+      url = user.getUris().get(ALBUMS_KEY).getUri();
+    }
+    return makeRequest(url, new TypeReference<SmugMugResponse<SmugMugAlbumsResponse>>() {})
+        .getResponse();
   }
 
-  SmugMugResponse<SmugMugUserResponse> makeUserRequest() throws IOException {
-    return makeRequest(USER_URL, new TypeReference<SmugMugResponse<SmugMugUserResponse>>() {});
+  /* Creates an album with albumName provided. */
+  SmugMugAlbumResponse createAlbum(String albumName) throws IOException {
+    // Set up album
+    Map<String, String> json = new HashMap<>();
+    String niceName = "Copy-of-" + albumName.replace(' ', '-');
```

Comment:
Why copy-of? Shouldn't we use the same album name as the original? Unless we are copying to the same account, i guess..

Reply:
This is what was done in the FlickrImporter too (and I recall also in google photos before I removed album support).

Reply:
Yeah, from other migration projects we have experience with, it functions as a way of distinguishing imported content. Another approach is to put all content in a new top level album or tag each one.